prow/kube/config_test.go

```diff
@@ -109,3 +116,152 @@ func TestMergeConfigs(t *testing.T) {
+				},
+				"default": {
+					Host:        "https://api.ci.l2s4.p1.openshiftapps.com:6443",
+					BearerToken: "REDACTED",
+				},
+				"hive": {
+					Host:        "https://api.hive.9xw5.p1.openshiftapps.com:6443",
+					BearerToken: "REDACTED",
+				},
+			},
+		},
+		{
+			name: "load from kubeconfigDir",
+			expected: map[string]rest.Config{
+				"": {
+					Host:        "https://api.build02.gcp.ci.openshift.org:6443",
+					BearerToken: "REDACTED",
+				},
+				"app.ci": {
+					Host:        "https://api.ci.l2s4.p1.openshiftapps.com:6443",
+					BearerToken: "REDACTED",
+				},
+				"build01": {
+					Host:        "https://api.build01.ci.devcluster.openshift.com:6443",
+					BearerToken: "REDACTED",
+				},
+				"build02": {
+					Host:        "https://api.build02.gcp.ci.openshift.org:6443",
+					BearerToken: "REDACTED",
+				},
+				"default": {
+					Host:        "https://api.build02.gcp.ci.openshift.org:6443",
+					BearerToken: "REDACTED",
+				},
+				"hive": {
+					Host:        "https://api.hive.9xw5.p1.openshiftapps.com:6443",
+					BearerToken: "REDACTED",
+				},
+			},
+		},
+		{
+			name:        "load from kubeconfigDir having contexts with the same name",
+			expectedErr: true,
+		},
+		{
+			name: "load from kubeconfig and kubeconfigDir",
+			expected: map[string]rest.Config{
+				"": {
+					Host:        "https://api.build01.ci.devcluster.openshift.com:6443",
+					BearerToken: "REDACTED",
+				},
+				"app.ci": {
+					Host:        "https://api.ci.l2s4.p1.openshiftapps.com:6443",
+					BearerToken: "REDACTED",
+				},
+				"build01": {
+					Host:        "https://api.build01.ci.devcluster.openshift.com:6443",
+					BearerToken: "REDACTED",
+				},
+				"build02": {
+					Host:        "https://api.build02.gcp.ci.openshift.org:6443",
+					BearerToken: "REDACTED",
+				},
+				"default": {
+					Host:        "https://api.build01.ci.devcluster.openshift.com:6443",
+					BearerToken: "REDACTED",
+				},
+				"hive": {
+					Host:        "https://api.hive.9xw5.p1.openshiftapps.com:6443",
+					BearerToken: "REDACTED",
+				},
+			},
+		},
+		{
+			name:     "load from kubeconfigDir with a filter",
+			regExRaw: `^sa.hook.*.config$`,
+			expected: map[string]rest.Config{
+				"": {
+					Host:        "https://api.ci.openshift.org:443",
+					BearerToken: "REDACTED",
+				},
+				"api.ci": {
+					Host:        "https://api.ci.openshift.org:443",
+					BearerToken: "REDACTED",
+				},
+				"default": {
+					Host:        "https://api.ci.openshift.org:443",
+					BearerToken: "REDACTED",
+				},
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			var kubeconfig string
+			var kubeconfigDir string
+			var projectedTokenFile string
+			for _, suffix := range []string{"kubeconfig", "kubeconfigDir", "projectedTokenFile"} {
+				fixtureName := filepath.Join("testdata", fmt.Sprintf("%s_%s", strings.ReplaceAll(t.Name(), "/", "_"), suffix))
+				if _, err := os.Stat(fixtureName); !os.IsNotExist(err) {
```

Comment:
why is it not an error for the file not to exist?

Reply:
Removed.